shared/actions/gregor.js

```diff
@@ -63,13 +63,15 @@ const setupEngineListeners = () => {
 
   // Filter this firehose down to the two systems we care about: "git", and "kbfs.favorites"
   // If ever you want to get OOBMs for a different system, then you need to enter it here.
-  RPCTypes.delegateUiCtlRegisterGregorFirehoseFilteredRpcPromise({systems: ['git', 'kbfs.favorites']})
-    .then(response => {
-      logger.info('Registered gregor listener')
-    })
-    .catch(error => {
-      logger.warn('error in registering gregor listener: ', error)
-    })
+  engine().actionOnConnect('registerGregorFirehose', () =>
+    RPCTypes.delegateUiCtlRegisterGregorFirehoseFilteredRpcPromise({systems: ['git', 'kbfs.favorites']})
+      .then(response => {
+        logger.info('Registered gregor listener')
+      })
+      .catch(error => {
+        logger.warn('error in registering gregor listener: ', error)
+      })
+  )
```

Comment:
what about the rpc call below?

Reply:
Ah, had to think a bit to see what to do about the call below. I think you want to put this inside the callback passed to `actionOnConnect`:

```
// The startReachability ...
const action = RPCTypes.reachabilityStartReachabilityRpcPromise()
  .then(reachability => GregorGen.createUpdateReachability({reachability}))
  .catch(err => {
    logger.warn('error bootstrapping reachability: ', err)	         logger.warn('error bootstrapping reachability: ', err)
  })
  return action	
```

`actionOnConnect` dispatches any action returned by the callback, so on connect, if this 2nd RPC succeeds, then it'll return a `createUpdateReachability` action, which will then get dispatched.

Then this outer function can just not return anything, since there's nothing to do until the connect callback.

Reply:
(also please fix the typo in the comment -- `startReachi` -> `startReacha`)